javascript/modules/client/src/client-erc20.ts

```diff
@@ -434,21 +432,140 @@ export class ClientErc20 extends ClientCore implements IClientErc20 {
     );
   }
 
-  private _getProposal(proposalId: string): Promise<Erc20Proposal> {
+  private async _getProposal(
+    proposalId: string,
+  ): Promise<Erc20Proposal | null> {
     if (!proposalId) {
       throw new Error("Invalid proposalId");
     }
-    const proposal = getDummyErc20Proposal(proposalId);
-    proposal.status = getProposalStatus(
-      proposal.startDate,
-      proposal.endDate,
-      true,
-      proposal.result.yes,
-      proposal.result.no,
-    );
-    return new Promise((resolve) => setTimeout(resolve, 1000)).then(
-      () => (proposal),
-    );
+    try {
+      await this.graphql.ensureOnline();
+      const client = this.graphql.getClient();
+      const { erc20VotingProposal } = await client.request(QueryErc20Proposal, {
+        proposalId,
+      });
+      if (!erc20VotingProposal) {
+        return null;
+      }
+      // TODO
+      // delete this cid once the proposals in subgraph have the correct
+      // format in the metadata field
+      const test_cid = "QmXhJawTJ3PkoKMyF3a4D89zybAHjpcGivkb7F1NkHAjpo";
+      const metadataString = await this.ipfs.fetchString(test_cid);
+      const metadata = JSON.parse(metadataString) as ProposalMetadata;
+      const startDate = new Date(
+        parseInt(erc20VotingProposal.startDate) * 1000,
+      );
+      const endDate = new Date(parseInt(erc20VotingProposal.endDate) * 1000);
+      const creationDate = new Date(
+        parseInt(erc20VotingProposal.createdAt) * 1000,
+      );
+      let usedVotingWeight: bigint = BigInt(0);
+      erc20VotingProposal.voters.forEach(
+        (voter: { voter: { id: string }; vote: string; weight: string }) => {
+          usedVotingWeight = usedVotingWeight + BigInt(voter.weight);
```

Comment:
```suggestion
          usedVotingWeight += BigInt(voter.weight);
```

Reply:
I would either use a plain `for` loop, or do a `reduce()` instead of a `forEach()`